### PR TITLE
sim integration prep

### PR DESCRIPTION
**Pull request by adamfk**

### CODE REVIEW

File: test-gen-2/src/LightSm.csx [hunk 1]
```diff
@@ -433,11 +571,19 @@ class MermaidGenerator : IVertexVisitor
...
         string initialStateId = MakeVertexDiagramId(initialState);
 
-        // Mermaid and PlantUML don't have a syntax that allows transitions to an initial state.
-        // If you do `someState --> [*]`, it means transition to a final state.
-        // StateSmith, however, does allow transitions to initial states so we add a dummy state to represent the initial state.
-        AppendLn($"[*] --> {initialStateId}");
-        mermaidEdgeTracker.AdvanceId();  // we skip this "work around" edge for now. We can improve this later.
+        bool showInitialStateDot = false;
+        // disabled for now because we then need to change edge highlighting.
+        // @emmby: can we we potentially style the `$initial_state` so that it looks somewhat like an initial state black filled circle? Thoughts?
```
Comment (emmby) [inline test-gen-2/src/LightSm.csx:576]: we could easily use a classdef to define a css style, and then stick whatever CSS we want to style that class

File: test-gen-2/src/LightSm.csx [hunk 2]
```diff
@@ -349,19 +393,113 @@ void LoggingTransformationStep(StateMachine sm)
...
+        // GIL is Generic Intermediary Language. It is used by history vertices and other special cases.
+        if (behavior.actionCode.Contains("$gil("))
+        {
+            // keep actual code
+            behavior.actionCode += $"""addHistoryRow(new Date(), "Executed action: " + {EscapeFsmCode(behavior.actionCode)});""";
+        }
+        else
+        {
+            // we don't want to execute the action, just log it.
+            behavior.actionCode = $"""addHistoryRow(new Date(), "FSM would execute action: " + {EscapeFsmCode(behavior.actionCode)});""";
```
[the remark this answers is not in the document]
Comment (adamfk) [reply test-gen-2/src/LightSm.csx:469]: Good call. Will do. I'm happy to see more improvements here. The only real important point in this jack hammer POC is that it works with any user diagram with no user changes.

File: test-gen-2/src/LightSm.csx (hunk 1 above)
Comment (adamfk) [reply test-gen-2/src/LightSm.csx:578]: I added to here: https://github.com/StateSmith/StateSmith/issues/294

File: test-gen-2/src/LightSm.csx (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamfk) [reply test-gen-2/src/LightSm.csx]: Thanks for mentioning this. It helped me remember a dirty FIXME I had in the POC: https://github.com/StateSmith/StateSmith-examples/pull/7/commits/78b83a1a1cf7a8587e97c5bc11b5080e9942532b#diff-490d4b13418f853c85403676ffde7773e123ded53b49d48822f7dd98d4050cb2L305

**Pull request merged**